src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentDetector.cs

```diff
@@ -109,25 +104,26 @@ private NpmAuthor GetAuthor(JToken authorToken, string packageName, string fileP
                 return null;
             }
 
-            string authorName = null;
+            string authorName;
             string authorEmail = null;
+            string authorSingleStringPattern = @"^([^<(]+?)?[ \t]*(?:<([^>(]+?)>)?[ \t]*(?:\(([^)]+?)\)|$)";
 
-            //for parsing author in Json Format
+            // for parsing author in Json Format
             // for e.g. 
             // "author": {
             //     "name": "John Doe",
             //     "email": "johndoe@outlook.com",
             //     "name": "https://jd.com",
             // }
-            if (authorString.StartsWith("{"))
+            if (authorToken.HasValues)
             {
                 authorName = authorToken["name"]?.ToString();
                 authorEmail = authorToken["email"]?.ToString();
             }
 
             // for parsing single line author.
             // for e.g. "author": "John Doe <johnDoe@outlook.com> (https://jd.com)"
-            else
+            else if (Regex.IsMatch(authorString, authorSingleStringPattern))
             {
                 bool isEmailPresent = authorString.Contains("<");
```

Comment:
You changed to regex matching, but then you still do string matching below. I think it's easier to extract match groups instead.

Reply:
fixed: using groups to recognize the email and name.